app/policies/case/foi/standard_policy.rb

```diff
@@ -8,7 +8,19 @@ def initialize(user, scope)
 
     def resolve
       if @user.permitted_correspondence_types.include? CorrespondenceType.foi
-        @scope
+        
+        if @user.responder_only?
+          team_restriction_sql = "select case_id from assignments 
+          join teams_users_roles 
+          on assignments.team_id = teams_users_roles.team_id
+          where teams_users_roles.role = '%<role>s' 
+          and teams_users_roles.user_id = %<user_id>i"
+
+          @scope.where(" cases.id in 
+          (#{team_restriction_sql}) " % {role: :responder.to_s, user_id: @user.id})
+        else
```

Comment:
This still makes me nervous - I think there is probably a way to have `ActiveRecord` construct this query.

See: http://joshfrankel.me/blog/constructing-a-sql-select-from-subquery-in-activerecord/